bofire/surrogates/robust_single_task_gp.py

```diff
@@ -0,0 +1,146 @@
+    def _fit(self, X: pd.DataFrame, Y: pd.DataFrame, **kwargs):
+        scaler = get_scaler(self.inputs, self.input_preprocessing_specs, self.scaler, X)
+        transformed_X = self.inputs.transform(X, self.input_preprocessing_specs)
+
+        tX, tY = (
+            torch.from_numpy(transformed_X.values).to(**tkwargs),
+            torch.from_numpy(Y.values).to(**tkwargs),
+        )
+
+        self.model = RobustRelevancePursuitSingleTaskGP(
+            train_X=tX,
+            train_Y=tY,
+            covar_module=kernels.map(
+                self.kernel,
+                batch_shape=torch.Size(),
+                active_dims=list(range(tX.shape[1])),
+                ard_num_dims=1,  # this keyword is ignored
+                features_to_idx_mapper=lambda feats: self.inputs.get_feature_indices(
+                    self.input_preprocessing_specs, feats
+                ),
+            ),
+            outcome_transform=(
+                Standardize(m=tY.shape[-1])
+                if self.output_scaler == ScalerEnum.STANDARDIZE
+                else None
+            ),
+            input_transform=scaler,
+            convex_parameterization=self.convex_parametrization,
+            cache_model_trace=self.cache_model_trace,
+        )
+        if self.prior_mean_of_support is not None:
+            self.model.prior_mean_of_support = self.prior_mean_of_support
+        self.model.likelihood.noise_covar.noise_prior = priors.map(self.noise_prior)
+        self.model.covar_module.lengthscale_constraint = priors.map(
+            self.lengthscale_constraint
+        )
+
+        mll = ExactMarginalLogLikelihood(self.model.likelihood, self.model)
+        fit_gpytorch_mll(
+            mll,
+            # TODO: with the Decorator to _fit_rrp in botorch robust_relevance_pursuit.py, options and max_attempts do not work
+            # TODO: still find workaround for this
+            # options=self.training_specs,
+            # max_attempts=10,
+            reset_parameters=False,  # question is if we want to be able to set this ourselves or not?
+            relevance_pursuit_optimizer=backward_relevance_pursuit,  # based on the docs this is most robust but more expensive that forward
+        )
+
+    def predict_outliers(
+        self, experiments: pd.DataFrame, options: Optional[Dict] = None
+    ) -> pd.DataFrame:
+        """
+        Predict the outliers for the given experiments. This is done by fitting the RobustSingleTaskGPSurrogate model and then using the model to do
+        predictions and obtain the data-point specific noise level (rho).
+        """
+        options = options or {}
+
+        # (re)fit model can only predict rho values for the data the model is fitted on
+        self.fit(experiments=experiments, **options)
+
+        # get model predictions, this should do a lot of validation, so we don't need it for the rhos.
+        predictions = self.predict(experiments)
+
+        # get the datapoint specific noise level
```

Comment:
`assert isinstance(self.model, RobustRelevancePursuitSingleTaskGP)` will solve the typing issue below